maro/simulator/scenarios/finance/common/common.py

```diff
@@ -0,0 +1,247 @@
+            commission: float, tax: float
+    ):
+        self.trade_number = int(trade_number)
+        self.price_per_item = price_per_item
+        self.commission = commission
+        self.tax = tax
+
+    @property
+    def total_cost(self):
+        return self.trade_number * self.price_per_item + self.commission + self.tax
+
+    def __repr__(self):
+        return f"<  \
+            number: {self.trade_number} price: {self.price_per_item} \
+            commission: {self.commission} tax: {self.tax} >"
+
+
+class DecisionEvent:
+    def __init__(
+        self, tick: int, item: int = -1,
+        action_scope_func: Callable = None, action_type: ActionType = ActionType.order
+    ):
+        """
+        Parameters:
+            tick (int): current tick of decision
+            item (int): available item index for action, such as a stock for StockSubEngine
+            action_scope_func (Callable): function to provide action scope
+            action_type (ActionType): type of the expected action
+        """
+        self.tick = tick
+        self.action_type = action_type
+        self.item = item
+        self._action_scope = None
+        self._action_scope_func = action_scope_func
+
+    @property
+    def action_scope(self):
+        """Action scope for items"""
+        if self._action_scope is None:
+            self._action_scope = self._action_scope_func(self.action_type, self.item, self.tick)
+
+        return self._action_scope
+
+    def __repr__(self):
+        return f"<DecisionEvent type: {self.action_type}, tick: {self.tick}>"
+
+
+class OrderActionScope:
+    def __init__(self, buy_min, buy_max, sell_min, sell_max, supported_order):
+        self.buy_min = buy_min
+        self.buy_max = buy_max
+        self.sell_min = sell_min
+        self.sell_max = sell_max
+        self.supported_order = supported_order
+
+
+class CancelOrderActionScope:
+    def __init__(self, available_orders):
+        self.available_orders = available_orders
+
+
+class Action(ABC):
+    idx = 0
+
+    def __init__(
+        self,
+        tick: int = 0,
+        id: int = None, life_time: int = 1
+    ):
+        """
+        Parameters:
+            item_index (int): index of the item (such as stock index), usually from DecisionEvent.items
+            number (int): number to perform, positive means buy, negitive means sell
+        """
+        self.decision_tick = tick
+        self.finish_tick = None
+        self.state = ActionState.pending
+        if id is not None:
+            self.id = id
+        else:
+            self.id = Action.idx
+            Action.idx += 1
+        self.life_time = life_time
+        self.action_result = None
+        self.comment = ""
+        # print("Action id:", self.id)
+
+    def __repr__(self):
+        return f"< Action start: {self.decision_tick} finished: {self.finish_tick} state: {self.state} >"
+
+
+class CancelOrder(Action):
+    def __init__(self, action, tick):
```

Comment:
What is the type of ``action`` here?

Add type for parameters. Same to other class/function